subprojects/execution/src/test/groovy/org/gradle/internal/execution/steps/StepSpec.groovy

```diff
@@ -21,13 +21,14 @@ import org.gradle.internal.operations.BuildOperationType
 import org.gradle.internal.operations.TestBuildOperationExecutor
 import org.gradle.test.fixtures.file.TestFile
 import org.gradle.test.fixtures.file.TestNameTestDirectoryProvider
-import org.junit.Rule
+import org.junit.ClassRule
+import spock.lang.Shared
 import spock.lang.Specification
 
 import java.util.function.Consumer
 
 abstract class StepSpec<C extends Context> extends Specification {
-    @Rule
+    @Shared @ClassRule
```

Comment:
Why did you convert this into a shared class rule?

Reply:
So that we can access it from `where:`. I have a better idea how to do this that doesn't require `cleanup()`, might get around doing it.

Reply:
I didn't see it accessed from `where:`.